pages/blog/_posts/2020-03-30-catch-imsi-from-vowifi-session.md

```diff
@@ -0,0 +1,157 @@
+will secure and encrypt both the data and maintain the integrity of the
+connection throughout the session.
+
+The UE finds the epdg termination point by looking dns records partly following
+a convention decided by 3gpp and typically looks like this in DNS:
+
+```
+epdg.epc.mnc001.mcc242.pub.3gppnetwork.org. 3488 IN A 77.16.1.126
+epdg.epc.mnc001.mcc242.pub.3gppnetwork.org. 3488 IN A 77.16.1.125
+```
+
+From the DNS records we recognize the network operator and and the country code,
+in case corresponds to Norway (242) and Telenor (001).
+
+The DNS records are registered under under a delegated domain owned by GSMA and
+usually are redelegated to operator under their own umbrella like the example
+mnc001.mcc242.pub.3gppnetwork.org.
+
+## The problem
+When UE establish session to epdg it uses a vpn, a ipsec relationship using
+IKEv2 for authentication, encryption and integrity.
+
+So far the implementation works as intended and provides good security through
+encryption and security.
+
+The problem is not the VoWifi per see, by rather how ipsec establish the
+session. When UE connects to the epdg, it acts as a initiator and the epdg is
+inherently passive since it cannot know where from (ip) the UE will come from.
+
+![sim-aka flow](/img/blog/vowifi-imsi-leak/sim-aka.png)
+
+### EAP-AKA exposes identity
+Vowifi as mentioned earlier utilises an encryption protocol based on the widely
+adopted Extensible Authentication Protocol. EAP itself is just a protocol and
+does not define the contents of the data or how exact the data exchanges look
+like. EAP-AKA unfortunately exposes user identity in the clear unencrypted during
+the authentication session and in this case the user identity is equal to the
+imsi.
+
+#### Solution
+This is the hardest problem to solve since it needs a security layer or settings
+that comes before ipsec starts to connect.
+
+The proposed solutions
+* Force the use of conservative peer for eap-aka/sim
+and use pseudonym identity (tmsi) to avoid exposing imsi.
+* Enable EAP-TTLS in addition to EAP-AKA/SIM
+* Only connect trusted/encrypted AP's
+
+### Fake Ipsec termination exposes identity
+By impersonating a epdg by redirecting all dns requests for any
+`pub.3gppnetwork.org.` to our own fake ipsec termination providing just enough
+to catch the imsi.
+
+A raspberry pie can easily be setup to constantly scan for open wifi networks,
+then impersonating the ssid in hope of lure ue's to connect. Any UE set to use
+VoWiFi connecting to the fake access point will give away their imsi.
+
+A wifi ssid scan example of what we can automate. In this case the Isfjell-Guest
+would have been picked to catch imsi since its open and unencrypted:
+
+![wifi-ssid](/img/blog/vowifi-imsi-leak/wlan-ssid.png)
+
+Snippet from the ipsec termination, UE (iPhone 8) exposes imsi several times:
+```
+13[ENC] parsed IKE_AUTH request 2 [ EAP/RES/AKA ]
+13[IKE] '0242014511******@wlan.mnc001.mcc242.3gppnetwork.org' is not a reauth identity
```

Comment:
Replace IMSI with 09999994511**** and MNC MCC

Reply:
Done